docarray/base_document/document.py

```diff
@@ -1,24 +1,39 @@
 import os
-from typing import Type
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Optional, Type
 
 import orjson
-from pydantic import BaseModel, Field, parse_obj_as
+from pydantic import BaseModel, Field, PrivateAttr, parse_obj_as
 from rich.console import Console
 
 from docarray.base_document.base_node import BaseNode
 from docarray.base_document.io.json import orjson_dumps, orjson_dumps_and_decode
 from docarray.base_document.mixins import IOMixin, UpdateMixin
 from docarray.typing import ID
 
+if TYPE_CHECKING:
+    from docarray.array.stacked.array_stacked import DocumentArrayStacked
+
+
 _console: Console = Console()
 
+object_setattr = object.__setattr__
+
+
+@dataclass
+class _DaRef:
+    ref: 'DocumentArrayStacked'
```

Comment:
I would call this `parent_da` or something like that instead of `ref`, which is very non-descript